tools/@aws-cdk/cli-args-gen/lib/yargs-gen.ts

```diff
@@ -116,7 +113,13 @@ function makeYargs(config: CliConfig, helpers: CliHelpers): Statement {
 function makeOptions(prefix: Expression, options: { [optionName: string]: CliOption }, helpers: CliHelpers) {
   let optionsExpr = prefix;
   for (const option of Object.keys(options)) {
-    const theOption: CliOption = options[option];
+    const theOption: CliOption = {
+      // Make the default explicit (overridden if the option includes an actual default)
+      // 'notification-arns' is a special snowflake that should be defaulted to 'undefined', but https://github.com/yargs/yargs/issues/2443
+      // prevents us from doing so. This should be changed if the issue is resolved.
+      ...(option === 'notification-arns' ? {} : { default: generateDefault(options[option].type) }),
+      ...options[option],
+    };
```

Comment:
https://github.com/yargs/yargs/issues/2443 -> this issue prevents me from specifying and undefined default

https://github.com/aws/aws-cdk/blob/61626dc9e5d580878dec665aa2d2ac976103a826/packages/aws-cdk/lib/cdk-toolkit.ts#L374-L378 -> this is relevant for why `notification-arns` is a special case.